services/filemanager/digitalOceanSpacesManager.go

```diff
@@ -0,0 +1,197 @@
+package filemanager
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	SpacesManager "github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+// Upload passed in file to spaces
+func (manager *DOSpacesManager) Upload(file *os.File, prefixes ...string) (UploadOutput, error) {
+	if manager.Config.Bucket == "" {
+		return UploadOutput{}, errors.New("no storage bucket configured to uploader")
+	}
+
+	region := misc.GetSpacesLocation(manager.Config.EndPoint)
+	var uploadSession *session.Session
+	if manager.Config.AccessKeyID == "" || manager.Config.AccessKey == "" {
+		uploadSession = session.New(&aws.Config{
+			Credentials: credentials.NewStaticCredentials(manager.Config.AccessKeyID, manager.Config.AccessKey, ""),
+			Region:      aws.String(region),
+			Endpoint:    aws.String(manager.Config.EndPoint),
+		})
+	} else {
+		uploadSession = session.New(&aws.Config{
+			Region:      aws.String(region),
+			Credentials: credentials.NewStaticCredentials(manager.Config.AccessKeyID, manager.Config.AccessKey, ""),
+			Endpoint:    aws.String(manager.Config.EndPoint),
+		})
+	}
+
+	s3Client := s3.New(uploadSession)
+	splitFileName := strings.Split(file.Name(), "/")
+	fileName := ""
+	if len(prefixes) > 0 {
+		fileName = strings.Join(prefixes[:], "/") + "/"
+	}
+	fileName += splitFileName[len(splitFileName)-1]
+	if manager.Config.Prefix != "" {
+		if manager.Config.Prefix[len(manager.Config.Prefix)-1:] == "/" {
+			fileName = manager.Config.Prefix + fileName
+		} else {
+			fileName = manager.Config.Prefix + "/" + fileName
+		}
+	}
+	uploadInput := s3.PutObjectInput{
+		ACL:    aws.String("bucket-owner-full-control"),
+		Bucket: aws.String(manager.Config.Bucket),
+		Key:    aws.String(fileName),
+		Body:   file,
+	}
+	if manager.Config.EnableSSE {
+		uploadInput.ServerSideEncryption = aws.String("AES256")
+	}
+	_, err := s3Client.PutObject(&uploadInput)
+	if err != nil {
+		return UploadOutput{}, err
+	}
+	var location string
```

Comment:
remove var and do location :=

Reply:
does s3manager return bucket name in output location. checking because we store this staging file location in warehouse and use it to download file